[
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/411883354",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-411883354",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 411883354,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMTg4MzM1NA==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-09T20:16:58Z",
    "updated_at": "2018-08-09T20:17:26Z",
    "author_association": "MEMBER",
    "body": "@jl0000285 : Thanks for the report!  I'm confused here: when you say, \"cause errors whenever used with any other library\" - can you provide examples to illustrate your point?\r\n\r\nSecond, IIUC, `DataFrame` can't hold Python `int` as a numeric dtype.  It will always end up using a `numpy` dtype to hold such data."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/411887422",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-411887422",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 411887422,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMTg4NzQyMg==",
    "user": {
      "login": "TomAugspurger",
      "id": 1312546,
      "node_id": "MDQ6VXNlcjEzMTI1NDY=",
      "avatar_url": "https://avatars3.githubusercontent.com/u/1312546?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/TomAugspurger",
      "html_url": "https://github.com/TomAugspurger",
      "followers_url": "https://api.github.com/users/TomAugspurger/followers",
      "following_url": "https://api.github.com/users/TomAugspurger/following{/other_user}",
      "gists_url": "https://api.github.com/users/TomAugspurger/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/TomAugspurger/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/TomAugspurger/subscriptions",
      "organizations_url": "https://api.github.com/users/TomAugspurger/orgs",
      "repos_url": "https://api.github.com/users/TomAugspurger/repos",
      "events_url": "https://api.github.com/users/TomAugspurger/events{/privacy}",
      "received_events_url": "https://api.github.com/users/TomAugspurger/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-09T20:32:11Z",
    "updated_at": "2018-08-09T20:32:11Z",
    "author_association": "CONTRIBUTOR",
    "body": "I wonder if in https://github.com/pandas-dev/pandas/blob/475e391eaa0c85c4615ad6da5f81dee9b890635b/pandas/io/sql.py#L884-L886\r\n\r\nthat should be `np.dtype(int).type`?"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/412005683",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-412005683",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 412005683,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMjAwNTY4Mw==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-10T07:52:57Z",
    "updated_at": "2018-08-10T07:52:57Z",
    "author_association": "MEMBER",
    "body": "@TomAugspurger : That will still return a `numpy` integer dtype, not a native Python one, which is what I *think* @jl0000285 is concerned about."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/412070359",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-412070359",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 412070359,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMjA3MDM1OQ==",
    "user": {
      "login": "TomAugspurger",
      "id": 1312546,
      "node_id": "MDQ6VXNlcjEzMTI1NDY=",
      "avatar_url": "https://avatars3.githubusercontent.com/u/1312546?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/TomAugspurger",
      "html_url": "https://github.com/TomAugspurger",
      "followers_url": "https://api.github.com/users/TomAugspurger/followers",
      "following_url": "https://api.github.com/users/TomAugspurger/following{/other_user}",
      "gists_url": "https://api.github.com/users/TomAugspurger/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/TomAugspurger/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/TomAugspurger/subscriptions",
      "organizations_url": "https://api.github.com/users/TomAugspurger/orgs",
      "repos_url": "https://api.github.com/users/TomAugspurger/repos",
      "events_url": "https://api.github.com/users/TomAugspurger/events{/privacy}",
      "received_events_url": "https://api.github.com/users/TomAugspurger/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-10T12:40:21Z",
    "updated_at": "2018-08-10T12:40:21Z",
    "author_association": "CONTRIBUTOR",
    "body": "I'm not quite sure what @jl0000285 is after, but we should be consistent with, e.g. `pd.Series(np.arange(10))`. Is that an int64 or int32 on Windows and Python2?"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/412149599",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-412149599",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 412149599,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMjE0OTU5OQ==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-10T17:22:56Z",
    "updated_at": "2018-08-10T17:22:56Z",
    "author_association": "MEMBER",
    "body": "@TomAugspurger : AFAIK, it could be either, depending on what `np.dtype(int).type` gives you."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/412156394",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-412156394",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 412156394,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMjE1NjM5NA==",
    "user": {
      "login": "jl0000285",
      "id": 16259960,
      "node_id": "MDQ6VXNlcjE2MjU5OTYw",
      "avatar_url": "https://avatars3.githubusercontent.com/u/16259960?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jl0000285",
      "html_url": "https://github.com/jl0000285",
      "followers_url": "https://api.github.com/users/jl0000285/followers",
      "following_url": "https://api.github.com/users/jl0000285/following{/other_user}",
      "gists_url": "https://api.github.com/users/jl0000285/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jl0000285/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jl0000285/subscriptions",
      "organizations_url": "https://api.github.com/users/jl0000285/orgs",
      "repos_url": "https://api.github.com/users/jl0000285/repos",
      "events_url": "https://api.github.com/users/jl0000285/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jl0000285/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-10T17:47:36Z",
    "updated_at": "2018-08-10T18:00:35Z",
    "author_association": "NONE",
    "body": "@gfyoung: Two places I've seen failure on windows builds are in the building of json graphs in networkx (json_graph.tree(G,root) where G is composed of dataframes from said pandas dataframe) and in database operations using objects from said dataframe (such as sqlalchemy/engine/defaults various cursor.execute(statement,parameters) statements where parameters are objects taken from a pandas dataframe). When the result of some operation that returns a pandas dataframe contains numpy.int64 objects on windows machines, these operations fail, because numpy.int64 objects are not numbers on windows machines (these same instructions work fine on macs and probably other unix systems as well)\r\n\r\n@TomAugspurger: The thing I was asking/hoping you guys could implement is automatically returning the proper numpy object dependant on which operating system a dataframe is being crafted on, that is to say if an operation is carried out on a windows machine, rather than returning dataframes with numpy.int64 objects in them, the dataframe would contain numpy.int32 objects (as these actually subclass int on windows machines and thus do not cause type errors when passed to another library). At the moment im handling the error by manual application of df.astype(numpy.int32, errors='ignore') when ever I need to use pandas on the windows version of what I am working on but feel like this should be something thats handled within pandas itself, thus the bug report. "
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/412160035",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-412160035",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 412160035,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMjE2MDAzNQ==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-10T18:01:04Z",
    "updated_at": "2018-08-10T18:01:04Z",
    "author_association": "MEMBER",
    "body": "@jl0000285 : Hmm...I see.  Would @TomAugspurger proposed patch [above](https://github.com/pandas-dev/pandas/issues/22263#issuecomment-411887422) work for you then?"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/412995364",
    "html_url": "https://github.com/pandas-dev/pandas/issues/22263#issuecomment-412995364",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/22263",
    "id": 412995364,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQxMjk5NTM2NA==",
    "user": {
      "login": "TomAugspurger",
      "id": 1312546,
      "node_id": "MDQ6VXNlcjEzMTI1NDY=",
      "avatar_url": "https://avatars3.githubusercontent.com/u/1312546?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/TomAugspurger",
      "html_url": "https://github.com/TomAugspurger",
      "followers_url": "https://api.github.com/users/TomAugspurger/followers",
      "following_url": "https://api.github.com/users/TomAugspurger/following{/other_user}",
      "gists_url": "https://api.github.com/users/TomAugspurger/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/TomAugspurger/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/TomAugspurger/subscriptions",
      "organizations_url": "https://api.github.com/users/TomAugspurger/orgs",
      "repos_url": "https://api.github.com/users/TomAugspurger/repos",
      "events_url": "https://api.github.com/users/TomAugspurger/events{/privacy}",
      "received_events_url": "https://api.github.com/users/TomAugspurger/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2018-08-14T19:54:42Z",
    "updated_at": "2018-08-14T19:54:42Z",
    "author_association": "CONTRIBUTOR",
    "body": "@jl0000285 thoughts on \r\nhttps://github.com/pandas-dev/pandas/issues/22263#issuecomment-412160035?\r\n\r\nCould you also edit your original post to include a full example that would be suitable for a unit test?"
  }
]
